product-service/src/test/java/com/org/project/shreyaa/ProductServiceApplicationTests.java

```diff
@@ -0,0 +1,72 @@
+package com.org.project.shreyaa;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.org.project.shreyaa.dto.ProductRequest;
+import com.org.project.shreyaa.repository.ProductRepository;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.DynamicPropertyRegistry;
+import org.springframework.test.context.DynamicPropertySource;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.testcontainers.containers.MongoDBContainer;
+import org.testcontainers.junit.jupiter.Testcontainers;
+
+import java.math.BigDecimal;
+
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+@Testcontainers
+class ProductServiceApplicationTests {
+	static final MongoDBContainer mongoDBContainer;
+
+	static {
+		mongoDBContainer = new MongoDBContainer("mongo:4.4.2");
+		mongoDBContainer.start();
+	}
+
+	@Autowired
+	private MockMvc mockMvc;
```

Comment:
mockmvc cannot be used in integration test. everything should be real except outgoing third party api ( using wiremocks) OR ( database using test containers). all other components must be real. Use TestRestTemplate to call the api